processor/compile/image.ts

```diff
@@ -0,0 +1,18 @@
+import type { Image } from "mdast";
+
+export default function ImageCompiler() {
+  const { Compiler } = this;
+  const { visitors } = Compiler.prototype;
+
+  const originalImageCompiler = visitors.image;
+
+  visitors.image = function compile(node: Image, ...args: any[]): string {
+    if (node.data?.hProperties?.className === 'emoji') return node.title;
+
+    const { align, className, width } = node.data?.hProperties || {};
+    const complexImage: boolean = Boolean(width) || Boolean(className) || Boolean(align);
+    if (complexImage) return `<Image ${{...node.data?.hProperties}} />`;
+
+    return originalImageCompiler.call(this, node, ...args);
+  };
```

Comment:
This is the old (`remark-stringify@8`) syntax. Here's an example of the a new-style plugin: https://github.com/readmeio/markdown/blob/beta/processor/compile/gemoji.ts#L1-L13

Also, my code-tabs PR refactors it a little: https://github.com/readmeio/markdown/pull/871/files#diff-3dd941d5e7f0c5b195b587d965c70f0e20f53da74c480b7d208e555b0dca2715R1-R12

Reply:
I guess I wasn't aware of this 3 weeks ago...

Reply:
hah! awesome--i had seen the syntax change but wasn't sure if that was just gemoji related

Reply:
It's like, well typed, but I could only figure it out by reading the remark source. 😖